database/dataGenerator.js

```diff
@@ -67,12 +69,123 @@ const createMenu = (id) => {
   return menu;
 };
 
+// Writes meals to a CSV file based on a generated menu.
+const mealsToCsv = (writer, quantity, updates) => {
+  const startTime = Date.now();
+  let countDown = quantity;
+  let estimatedTimeCalculated = false;
+  let estimatedTimeRemaining = 0;
+  let menuCount = 0;
+  const divisor = Math.floor(countDown / updates);
+  const header = 'meal_label,description,price,business_id,category_id,subcategory_id\n';
+  writer.write(header, 'utf8');
+
+  const write = () => {
+    let ok = true;
+    do {
+      const menu = createMenu();
+      if (countDown % divisor === 0 && menuCount > 0) {
+        const transpired = Date.now() - startTime;
+        if (!estimatedTimeCalculated) {
+          estimatedTimeRemaining = transpired * updates;
+          estimatedTimeCalculated = true;
+        }
+        if (transpired < 60000) {
```

Comment:
this code looks like it was reused in multiple locations, might be able to make this a function and call it